File: llm-platform-starter-sets/module-template/adapters/typescript/module.ts
/**
 * module.ts — TypeScript wrapper entrypoint for [your-module-name].
 *
 * This is the thin adapter between the platform runtime and the module's logic.
 * Keep this file clean. Heavy lifting goes in the tool adapters.
 *
 * Usage:
 *   import { run } from "./adapters/typescript/module";
 *   const result = await run(context);
 */

export interface ModuleContext {
  clientId: string;
  clientName: string;
  tone: string;
  moduleName: string;
  moduleVersion: string;
  userRequest: string;
  dataSources: string[];
  permissions: string[];
  outputFields: string[];
}

export interface ModuleOutput {
  summary: string;
  risks: string[];
  recommended_actions: string[];
  [key: string]: unknown; // Allow additional output fields
}

export async function run(context: ModuleContext): Promise<ModuleOutput> {
  /**
   * Main entrypoint for this module.
   *
   * Returns an object matching the fields defined in output_schema.json.
   * At minimum: { summary, risks, recommended_actions }
   */

  const { clientName, userRequest, dataSources, permissions } = context;

  // ── PHASE 2 STUB ──────────────────────────────────────────────────────────
  // Replace this section with real logic in Phase 5+.
  //
  // Phase 5 implementation pattern:
  //
  //   import { getDataExample, createRecordExample } from "../../src/tools/yourTools";
  //
  //   if (!permissions.includes("read")) {
  //     return errorResponse("Insufficient permissions for read");
  //   }
  //
  //   const data = await getDataExample({
  //     sourceId: dataSources[0] ?? "",
  //   });
  //
  //   const summary = deriveSummary(data, userRequest);
  //   const risks = identifyRisks(data);
  //   const actions = recommendActions(risks, permissions);
  //
  //   return { summary, risks, recommended_actions: actions };
  // ─────────────────────────────────────────────────────────────────────────

  return {
    summary: `[STUB] Module processed request for ${clientName}: ${userRequest.slice(0, 50)}`,
    risks: ["[STUB] Risk placeholder — connect real tools in Phase 5"],
    recommended_actions: ["[STUB] Action placeholder — connect real tools in Phase 5"],
  };
}

function errorResponse(message: string): ModuleOutput {
  return {
    summary: `Error: ${message}`,
    risks: ["Unable to process request"],
    recommended_actions: ["Review permissions and retry"],
  };
}

File: llm-platform-starter-sets/module-template/adapters/python/module.py
"""
module.py — Python wrapper entrypoint for [your-module-name].

This is the thin adapter between the platform runtime and the module's logic.
Keep this file clean. Heavy lifting goes in the tool adapters.

Usage:
    from adapters.python.module import run
    result = run(context)
"""

from typing import Any, Dict


def run(context: Dict[str, Any]) -> Dict[str, Any]:
    """
    Main entrypoint for this module.

    Args:
        context: ExecutionContext dict containing:
            - client_id (str)
            - client_name (str)
            - tone (str)
            - module_name (str)
            - module_version (str)
            - user_request (str)
            - data_sources (list[str])
            - permissions (list[str])
            - output_fields (list[str])

    Returns:
        Dict matching the output fields defined in output_schema.json.
        At minimum: { "summary": str, "risks": list, "recommended_actions": list }
    """

    user_request = context.get("user_request", "")
    client_name = context.get("client_name", "")
    data_sources = context.get("data_sources", [])
    permissions = context.get("permissions", [])

    # ── PHASE 2 STUB ───────────────────────────────────────────────────────────
    # Replace the section below with real logic in Phase 5+.
    #
    # Phase 5 implementation pattern:
    #
    #   from app.tools.your_tools import get_data_example, create_record_example
    #
    #   if "read" in permissions:
    #       data = get_data_example(source_id=data_sources[0] if data_sources else "")
    #   else:
    #       return error_response("Insufficient permissions for read")
    #
    #   # Process data, derive summary/risks/actions
    #   summary = derive_summary(data, user_request)
    #   risks = identify_risks(data)
    #   actions = recommend_actions(risks, permissions)
    #
    #   return {
    #       "summary": summary,
    #       "risks": risks,
    #       "recommended_actions": actions,
    #   }
    # ──────────────────────────────────────────────────────────────────────────

    return {
        "summary": f"[STUB] Module processed request for {client_name}: {user_request[:50]}",
        "risks": ["[STUB] Risk placeholder — connect real tools in Phase 5"],
        "recommended_actions": ["[STUB] Action placeholder — connect real tools in Phase 5"],
    }


def error_response(message: str) -> Dict[str, Any]:
    """Return a structured error output that still passes schema validation."""
    return {
        "summary": f"Error: {message}",
        "risks": ["Unable to process request"],
        "recommended_actions": ["Review permissions and retry"],
    }
